WordPress/Classes/Services/Stories/StoryMediaLoader.swift

```diff
@@ -52,7 +54,16 @@ final class StoryMediaLoader {
                 DDLogError("Error unarchiving \(file.url) - \(error)")
             }
 
-            service.getMediaWithID(NSNumber(value: Double(file.id) ?? 0), in: post.blog, success: { [weak self] media in
+            Task { @MainActor [weak self] in
+                let media: Media
+                do {
+                    let mediaID = try await mediaRepository.getMedia(withID: NSNumber(value: Double(file.id) ?? 0), in: blogID)
```

Comment:
Did you consider aborting if `Double(file.id)` is `nil`?

Or, is it safe to assume that there's no media with id 0 and therefore this fallback will always result in `getMedia` throwing `.mediaNotFound`?